## Tahap 1: Business Understanding

### 1.Menentukan Objektif Bisnis

Tujuan utama dari proyek ini adalah untuk memprediksi harga penjualan rumah (SalePrice) secara akurat.

Dengan model prediksi harga rumah yang akurat, diharapkan dapat membantu berbagai pihak yang terlibat dalam industri perumahan, seperti:


*   Pembeli rumah, untuk memperkirakan harga yang wajar untuk rumah yang mereka inginkan
*   Penjual rumah, untuk menentukan harga jual yang kompetitif.

*   Agen properti, pengembang, dan pihak lain, untuk pengambilan keputusan yang lebih baik.









### 2. Menetapkan Tujuan Teknis:



*   Mengembangkan model prediksi harga rumah yang akurat dengan menggunakan teknik machine learning, khususnya gradient boosting.
*   Mengidentifikasi fitur-fitur yang paling berpengaruh terhadap harga rumah.


*   Mengoptimalkan model untuk mencapai kinerja prediksi yang optimal, dengan metrik evaluasi RMSE (Root Mean Squared Error).


### 3. Membuat Rencana Proyek

#### Timeline proyek

## Data Understanding

### Mengumpulkan data

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
## Import data set
df = pd.read_csv("train.csv")

In [3]:
#menampilkan 10 data teratas
df.head(10)

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   
5   6          50       RL         85.0    14115   Pave   NaN      IR1   
6   7          20       RL         75.0    10084   Pave   NaN      Reg   
7   8          60       RL          NaN    10382   Pave   NaN      IR1   
8   9          50       RM         51.0     6120   Pave   NaN      Reg   
9  10         190       RL         50.0     7420   Pave   NaN      Reg   

  LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
5         Lvl    AllPub  ...        0    NaN  MnPrv        Shed     700   
6         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
7         Lvl    AllPub  ...        0    NaN    NaN        Shed     350   
8         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
9         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   

  MoSold YrSold  SaleType  SaleCondition  SalePrice  
0      2   2008        WD         Normal     208500  
1      5   2007        WD         Normal     181500  
2      9   2008        WD         Normal     223500  
3      2   2006        WD        Abnorml     140000  
4     12   2008        WD         Normal     250000  
5     10   2009        WD         Normal     143000  
6      8   2007        WD         Normal     307000  
7     11   2009        WD         Normal     200000  
8      4   2008        WD        Abnorml     129900  
9      1   2008        WD         Normal     118000  

[10 rows x 81 columns]

In [5]:
# drop kolom id karna tidak diperlukan
df = df.drop(columns=['Id'])

In [6]:
# menampilkan jumlah baris dan kolom
df.shape 

(1460, 80)

In [7]:
# menampilkan informasi data set
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 80 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   MSSubClass     1460 non-null   int64  
 1   MSZoning       1460 non-null   object 
 2   LotFrontage    1201 non-null   float64
 3   LotArea        1460 non-null   int64  
 4   Street         1460 non-null   object 
 5   Alley          91 non-null     object 
 6   LotShape       1460 non-null   object 
 7   LandContour    1460 non-null   object 
 8   Utilities      1460 non-null   object 
 9   LotConfig      1460 non-null   object 
 10  LandSlope      1460 non-null   object 
 11  Neighborhood   1460 non-null   object 
 12  Condition1     1460 non-null   object 
 13  Condition2     1460 non-null   object 
 14  BldgType       1460 non-null   object 
 15  HouseStyle     1460 non-null   object 
 16  OverallQual    1460 non-null   int64  
 17  OverallCond    1460 non-null   int64  
 18  YearBuil

Dari hasil keluaran diaas, kita bisa mendapat informasi bahwa data set memiliki jumlah kolom sebanyak 1460 dan 81 bari